src/swap/letsexchange.js

```diff
@@ -0,0 +1,268 @@
+            )
+
+      const {
+        fromMainnetCode,
+        toMainnetCode
+      } = getCodesWithMainnetTranscription(request, MAINNET_CODE_TRANSCRIPTION)
+
+      // Swap the currencies if we need a reverse quote:
+      const quoteParams = {
+        from: request.fromCurrencyCode,
+        to: request.toCurrencyCode,
+        network_from: fromMainnetCode,
+        network_to: toMainnetCode,
+        amount: quoteAmount
+      }
+
+      log('quoteParams:', quoteParams)
+
+      // Calculate the amounts:
+      let fromAmount, fromNativeAmount, toNativeAmount, reply
+      if (request.quoteFor === 'from') {
+        reply = await call(uri + 'info', request, {
+          params: quoteParams
+        })
+        fromAmount = quoteAmount
+        fromNativeAmount = request.nativeAmount
+        toNativeAmount = await request.toWallet.denominationToNative(
+          reply.amount.toString(),
+          request.toCurrencyCode
+        )
+      } else {
+        reply = await call(uri + 'info-revert', request, {
+          params: quoteParams
+        })
+        fromAmount = reply.amount
+        fromNativeAmount = await request.fromWallet.denominationToNative(
+          fromAmount.toString(),
+          request.fromCurrencyCode
+        )
+        toNativeAmount = request.nativeAmount
+      }
+
+      // Check the minimum:
+      const nativeMin = await request.fromWallet.denominationToNative(
+        reply.min_amount,
+        request.fromCurrencyCode
+      )
+
+      if (lt(fromNativeAmount, nativeMin)) {
+        throw new SwapBelowLimitError(swapInfo, nativeMin)
+      }
+
+      const { promoCode } = opts
+      const sendReply = await call(uri + 'transaction', request, {
+        params: {
+          deposit_amount: fromAmount,
+          coin_from: request.fromCurrencyCode,
+          coin_to: request.toCurrencyCode,
+          network_from: fromMainnetCode,
+          network_to: toMainnetCode,
+          withdrawal: toAddress,
+          return: fromAddress,
+          return_extra_id: null,
+          withdrawal_extra_id: null,
+          affiliate_id: initOptions.apiKey,
+          promocode: promoCode != null ? promoCode : '',
+          type: 'edge',
+          float: true,
+          isEstimate: false
+        }
+      })
+
+      log('sendReply', sendReply)
+      const quoteInfo = asQuoteInfo(sendReply)
+
+      // Make the transaction:
+      const spendInfo: EdgeSpendInfo = {
+        currencyCode: request.fromCurrencyCode,
+        spendTargets: [
+          {
+            nativeAmount: fromNativeAmount,
+            publicAddress: quoteInfo.deposit,
+            uniqueIdentifier: quoteInfo.deposit_extra_id
+          }
+        ],
+        networkFeeOption:
+          request.fromCurrencyCode.toUpperCase() === 'BTC'
+            ? 'high'
+            : 'standard',
+        swapData: {
+          orderId: quoteInfo.transaction_id,
+          orderUri: orderUri + quoteInfo.transaction_id,
+          isEstimate: false,
```

Comment:
Another note on floating and fixed exchanges, if LetsExchange supports different limits for each (like lower minimum amounts for floating rate) then it's worth attempting both types. For example, if LetsExchange can't server a fixed rate swap try again specifying a floating rate.

Reply:
Done!